packages/cmf-cqrs/src/middleware/smartWebsocket.js

```diff
@@ -54,15 +57,30 @@ export function startWebsocket(url, offlinebuffer, options) {
 		if (pingInterval) {
 			clearInterval(pingInterval);
 		}
+		if (pingTimeoutId) {
+			pingTimeoutId.close();
+		}
 	};
 	ws.onerror = function onerror(event) {
 		if (typeof onError === 'function') {
 			onError(event);
 		}
 	};
 	ws.ping = function ping() {
+		if (typeof onPing === 'function') {
+			if (!isNaN(options.pingTimeoutDelay)) {
```

Comment:
Why check for `NaN`? What else could it be?

Reply:
As it is provided by app parameters on createWebsocketMiddleware and that there is no strongly type, UI can't ensure that this is a Number, which if not the case, will lead provided parameter to be replaced by 0 and will spam the dispatch of timeout.

I like rather testing it and be defensive, that is far less confusing.